revocation/internal/crl/crl.go

```diff
@@ -153,7 +196,52 @@ func Supported(cert *x509.Certificate) bool {
 	return cert != nil && len(cert.CRLDistributionPoints) > 0
 }
 
-func validate(crl *x509.RevocationList, issuer *x509.Certificate) error {
+func hasFreshestCRL(extensions []pkix.Extension) bool {
+	return slices.ContainsFunc(extensions, func(ext pkix.Extension) bool {
+		return ext.Id.Equal(oidFreshestCRL)
+	})
+}
+
+func validate(bundle *crl.Bundle, issuer *x509.Certificate) error {
+	// validate base CRL
+	baseCRL := bundle.BaseCRL
+	if err := validateCRL(baseCRL, issuer); err != nil {
+		return fmt.Errorf("failed to validate base CRL: %w", err)
+	}
+
+	if bundle.DeltaCRL == nil {
+		return nil
+	}
+
+	// validate delta CRL
+	// RFC 5280, Section 5.2.4
+	deltaCRL := bundle.DeltaCRL
+	if err := validateCRL(deltaCRL, issuer); err != nil {
+		return fmt.Errorf("failed to validate delta CRL: %w", err)
+	}
+	if deltaCRL.Number.Cmp(baseCRL.Number) <= 0 {
+		return fmt.Errorf("delta CRL number %d is not greater than the base CRL number %d", deltaCRL.Number, baseCRL.Number)
+	}
+
+	// check delta CRL indicator extension
+	idx := slices.IndexFunc(deltaCRL.Extensions, func(ext pkix.Extension) bool {
+		return ext.Id.Equal(oidDeltaCRLIndicator)
+	})
+	if idx < 0 {
+		return errors.New("delta CRL indicator extension is not found")
+	}
```

Comment:
Same here, I'm seeing multiple places having this logic. Consider reducing the duplications.

Reply:
Updated.